tests/unit/test_zero_context.py

```diff
@@ -243,3 +243,42 @@ def test_ext_param_returnobj():
         assert len(net.dangler._external_params) == 0
         engine.backward(loss)
         engine.step()
+
+
+class ModelContainerVariableOutputType(ModelContainer):
+    def __init__(self, dim=16, output_type=dict):
+        super().__init__()
+        self.output_type = output_type
+        self.dim = dim
+        self.linear1 = torch.nn.Linear(dim, dim)
+
+    def forward(self, input):
+        act1 = self.linear1(input)
+        if self.output_type is None:
+            return None
+        if self.output_type is dict:
+            return {'loss': act1.sum()}
+        return act1.sum()
```

Comment:
It seems this code path will not be executed since the output types are restricted to dict and None?

Reply:
Yep totally right, I can remove this (wanted to keep it general)

Reply:
It is good to keep it general, then perhaps create a test case with a different output type?

Reply:
Done @tjruwase let me know what you think :)